release-24.2: sql/schemachanger: Exclude schema change wait time from idle session timeout

spilchen opened the pull request:
Backport 1/1 commits from #136463.

/cc @cockroachdb/release

---

Previously, the idle session timeout timer would start immediately after statement execution. If a transaction committed and had to wait for a schema change to complete, the connection could be interrupted if the idle session timeout expired during this wait.

This update pauses the idle session timer while waiting for schema change jobs to finish, ensuring connections remain active until the schema change completes.

Epic: None
Closes #135930
Release note (bug fix): The `idle_in_session_timeout` setting now excludes the time spent waiting for schema changer jobs to complete, preventing unintended session termination during schema change operations.
Release justification: low-risk bug that was reported by a customer using 23.2

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136502)
<!-- Reviewable:end -->

spilchen committed:
sql/schemachanger: Exclude schema change wait time from idle session timeout

Previously, the idle session timeout timer would start immediately after
statement execution. If a transaction committed and had to wait for a
schema change to complete, the connection could be interrupted if the
idle session timeout expired during this wait.

This update pauses the idle session timer while waiting for schema
change jobs to finish, ensuring connections remain active until the
schema change completes.

Epic: None
Closes #135930
Release note (bug fix): The `idle_in_session_timeout` setting now
excludes the time spent waiting for schema changer jobs to complete,
preventing unintended session termination during schema change
operations.

rafiss reviewed (approved):
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->
